pkg/performanceprofile/profilecreator/profilecreator.go

```
@@ -334,31 +334,53 @@ func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, sp
 }
 
 type cpuAccumulator struct {
-	builder *cpuset.Builder
-	count   int
+	elems map[int]struct{}
+	count int
+	done  bool
 }
 
 func newCPUAccumulator() *cpuAccumulator {
 	return &cpuAccumulator{
-		builder: cpuset.NewBuilder(),
+		elems: map[int]struct{}{},
+		count: 0,
+		done:  false,
 	}
 }
 
 // AddCores adds logical cores from the slice of *cpu.ProcessorCore to a CPUset till the cpuset size is equal to the max value specified
 // In case the max is specified as allCores, all the cores from the slice of *cpu.ProcessorCore are added to the CPUSet
-func (ca *cpuAccumulator) AddCores(max int, cores []*cpu.ProcessorCore) {
+func (ca *cpuAccumulator) AddCores(max int, cores []*cpu.ProcessorCore) int {
+	if ca.done {
+		return -1
```

let's return the error condition out of band:
```
func (ca *cpuAccumulator) AddCores(max int, cores []*cpu.ProcessorCore) (int, error) {
	if ca.done {
		return -1, fmt.Errorf("CPU accumulator finalized")
        }
        ...
}
```